geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java

```diff
@@ -945,17 +945,23 @@ protected void terminateDispatching(boolean checkQueue) {
     }
   }
 
-  private void closeSocket() {
-    if (this._socketClosed.compareAndSet(false, true)) {
+  private boolean closeSocket() {
+    String remoteHostAddress = this._remoteHostAddress;
+    if (this._socketClosed.compareAndSet(false, true) && remoteHostAddress != null) {
       // Close the socket
-      this._cacheClientNotifier.getSocketCloser().asyncClose(this._socket, this._remoteHostAddress,
-          null);
+      this._cacheClientNotifier.getSocketCloser().asyncClose(this._socket, remoteHostAddress, null);
       getCacheClientNotifier().getAcceptorStats().decCurrentQueueConnections();
+      return true;
     }
+    return false;
   }
 
   private void closeTransientFields() {
-    closeSocket();
+    if (!closeSocket()) {
```

Comment:
What happens if remoteHostAddress is null, we end up returning false from closeSocket and not doing the rest of this method.  Is that acceptable?